### PR TITLE
Handling missing days in a timeframe

### LINKED ISSUE #2924: Dashboard: Handling missing days in a timeframe
# Description
If a user has selected a particular timeframe, e.g. Last 7 days, they should see those 7 days in the charts. This should happen even if there are days without any data, which should be represented as empty columns.

# Reason
- Our charts should be truthful: there should be an equal distance between days
- Our chart should fulfil the user expectation: seeing all of the last 7 days

# Screenshot
dashboard-missing-ideas

### PR DESCRIPTION
**Pull request by marla-singer**
- Closes  #2924 
### Changes
- If a user has selected a particular timeframe, e.g. Last 7 days, then there display 7 days in the charts. 
- When there are days without any data then it is represented as empty columns

### CODE REVIEW

File: apinf_packages/dashboard/client/view/overview/charts/median_response_time/median_response_time.js [hunk 1]
```diff
@@ -67,15 +71,30 @@ Template.medianResponseTime.onRendered(function () {
...
     const elasticsearchData = Template.currentData().buckets;
 
-    // Get Labels for chart
-    const labels = elasticsearchData.map(value => {
+    // Get current timeframe
+    const dateCount = FlowRouter.getQueryParam('timeframe');
+
+    const params = {
+      // Date range must have equal length with dateCount value
+      // Today value is included then subtract less days
```
Comment (matleppa) [inline apinf_packages/dashboard/client/view/overview/charts/median_response_time/median_response_time.js]: I'd rather comment followingly
// To include also today, subtract one day less than count of days in period

File: apinf_packages/dashboard/client/view/overview/charts/requests_over_time/requests_over_time.js [hunk 2]
```diff
@@ -67,14 +71,30 @@ Template.requestsOverTime.onRendered(function () {
...
-      return moment(value.key).format('MM/DD');
-    });
+    // Get current timeframe
+    const dateCount = FlowRouter.getQueryParam('timeframe');
 
-    const data = elasticsearchData.map(value => {
-      return value.doc_count;
+    const params = {
+      // Date range must have equal length with dateCount value
+      // Today value is included then subtract less days
```
Comment (matleppa) [inline apinf_packages/dashboard/client/view/overview/charts/requests_over_time/requests_over_time.js]: I'd rather comment followingly
// To include also today, subtract one day less than count of days in period

File: apinf_packages/dashboard/client/view/overview/charts/unique_users/unique_users.js [hunk 3]
```diff
@@ -67,15 +71,30 @@ Template.uniqueUsersOverTime.onRendered(function () {
...
-    });
+    // Get current timeframe
+    const dateCount = FlowRouter.getQueryParam('timeframe');
 
-    // Get x & y points for chart
-    const data = elasticsearchData.map(value => {
-      return value.unique_users.buckets.length;
+    const params = {
+      // Date range must have equal length with dateCount value
+      // Today value is included then subtract less days
```
Comment (matleppa) [inline apinf_packages/dashboard/client/view/overview/charts/unique_users/unique_users.js]: I'd rather comment followingly
// To include also today, subtract one day less than count of days in period

**Comment by marla-singer** [comment]
@matleppa Update comments

**Pull request merged**